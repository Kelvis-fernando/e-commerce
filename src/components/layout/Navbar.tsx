import { Header, Logo, Links, Login } from "../../styles/layout/navbar";
import { ShoppingCart, Coffee } from "phosphor-react";
import { useModal } from "../../hooks/layout/useModal";
import { Link } from "react-router-dom";
import Modal from "./Modal";

const Navbar = () => {
const {setIsModalOpen, isModalOpen, handleModalClose} =  useModal()

  return (
    <Header>
      <Logo>
        <Link to="/home">
          <Coffee size={30} />
          <h1>Coffee Delivery</h1>
        </Link>
      </Logo>
      <Links>
        <Link to="/home">Home</Link>
        <Link to="/products">Produtos</Link>
        <Link to="/about">Sobre</Link>
        <Link to="/contact">Fale conosco</Link>
      </Links>
      <Login>
        <button onClick={() => setIsModalOpen(true)}>Login</button>
        <ShoppingCart size={30} />
      </Login>
      <Modal isOpen={isModalOpen} onClose={handleModalClose}>
        <h2>Modal</h2>
        <p>Este Ã© um exemplo de Modal.</p>
      </Modal>
    </Header>
  );
};

export default Navbar;
