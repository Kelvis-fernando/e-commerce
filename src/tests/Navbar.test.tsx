import React from "react";
import { render } from "@testing-library/react";
import Navbar from "../components/layout/Navbar";
import { MemoryRouter } from "react-router-dom";

describe("Navbar", () => {
  test("should render and match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
