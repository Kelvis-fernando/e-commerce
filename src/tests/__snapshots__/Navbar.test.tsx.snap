// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render and match snapshot 1`] = `
<div>
  <header
    class="sc-bcXHqe gasgdr"
  >
    <div
      class="sc-gswNZR ihVebz"
    >
      <a
        href="/home"
      >
        <svg
          fill="currentColor"
          height="30"
          viewBox="0 0 256 256"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="88"
            x2="88"
            y1="24"
            y2="56"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="120"
            x2="120"
            y1="24"
            y2="56"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="152"
            x2="152"
            y1="24"
            y2="56"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="32"
            x2="208"
            y1="216"
            y2="216"
          />
          <path
            d="M83.3,216A88,88,0,0,1,32,136V88H208v48a88,88,0,0,1-51.3,80"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M208,88h0a32,32,0,0,1,32,32v8a32,32,0,0,1-32,32h-3.4"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <h1>
          Coffee Delivery
        </h1>
      </a>
    </div>
    <div
      class="sc-dkrFOg dEFqbt"
    >
      <a
        href="/home"
      >
        Home
      </a>
      <a
        href="/products"
      >
        Produtos
      </a>
      <a
        href="/producer"
      >
        Sou produtor
      </a>
    </div>
    <div
      class="sc-eDvSVe bIROyA"
    >
      <button>
        Login
      </button>
      <div
        class="sc-fnGiBr dUiexO"
      >
        <svg
          fill="currentColor"
          height="30"
          viewBox="0 0 256 256"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M56.2,104a71.9,71.9,0,0,1,72.3-72c39.6.3,71.3,33.2,71.3,72.9V112c0,35.8,7.5,56.6,14.1,68a8,8,0,0,1-6.9,12H49a8,8,0,0,1-6.9-12c6.6-11.4,14.1-32.2,14.1-68Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M96,192v8a32,32,0,0,0,64,0v-8"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <div
          class="sc-bjfHbI hfrTsO"
        >
          <div>
            <h1>
              Pedidos
            </h1>
            <ul />
          </div>
        </div>
      </div>
      <div
        class="sc-fnGiBr dUiexO"
      >
        <svg
          fill="currentColor"
          height="30"
          viewBox="0 0 256 256"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M184,184H69.8L41.9,30.6A8,8,0,0,0,34.1,24H16"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="80"
            cy="204"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="184"
            cy="204"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M62.5,144H188.1a15.9,15.9,0,0,0,15.7-13.1L216,64H48"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
      </div>
    </div>
  </header>
</div>
`;
